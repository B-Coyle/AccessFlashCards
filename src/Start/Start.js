import React, {Component} from 'react';
import './Start.scss';

export default class Start extends Component {
    constructor(props) {
        super(props);
        this.state = { 
            selectedQuestion: [],
        }
    }



    startGame = (props) => {
      let randomQuestion = this.props.allQuestions.sort(() => .5 - Math.random()).pop();
      this.setState = {
        selectedQuestion: this.state.selectedQuestion.push(randomQuestion)
      }
      this.props.randomQuestion(this.state.selectedQuestion)
  }



    render(){
        return(
            <article>
                <input type="button" className="startBtn" onClick={this.startGame} value="Start Studying" />
            </article>
        )
    }
}