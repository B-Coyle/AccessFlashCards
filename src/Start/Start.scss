@import '../Styles/variables.scss';

.startBtn{
    border-radius: 5px;
    color: $fontColor;
    background-color: green;
    width: 220px;
    height: 50px;

}