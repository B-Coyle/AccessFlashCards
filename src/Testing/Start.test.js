import React from 'react';
import ReactDOM from 'react-dom';
import Start from '../Start/Start';
import { shallow } from 'enzyme'; 

describe('Footer', () => {
let wrapper;
let mockStart = jest.fn();
let mockQuestion = {
    "question": "What is accessibility?",
    "answer": "Is the design and creation of websites that can be used by everyone.",
    "id": 1,
    "category": "Definition"
  }


beforeEach(() => {
    wrapper = shallow(
        <Start
        selectedQuestion = {mockQuestion} />
    )
});

it('Should have default states', () => {
    expect(wrapper.state()).toEqual( {
      selectedQuestion: []
    })
  });

  it('Should match rendered snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})
