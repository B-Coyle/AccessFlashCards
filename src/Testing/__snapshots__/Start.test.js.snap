// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should match rendered snapshot 1`] = `undefined`;
